data/tribes/immovables/dike/init.lua

```diff
@@ -0,0 +1,26 @@
+push_textdomain("tribes")
+
+dirname = path.dirname (__file__)
+
+wl.Descriptions():new_immovable_type {
+   name = "dike",
+   -- TRANSLATORS: This is an immovable name used in lists of immovables
+   descname = pgettext ("immovable", "Dike"),
+   icon = dirname .. "menu.png",
+   size = "none",
+   programs = {
+      main = {
+         "animate=idle duration:10m",
```

Comment:
will test it but this seems to be to short as one diking step lasts 120 secs plus walking time. and we might need 5 * 2 steps for one dike to get arable land in the end. 
I thought for the dike more like a shipconstruction site where the diker dikes until all land is arable.

Reply:
Every time the worker interacts with the dike, there's a 33% chance it is removed from wear&tear, and if it survives then the lifetime is reset to the value specified in the second program. I doubled both times now.

A shipconstruction-like lifecycle is not really possible because there is no way to know how many triangles around a dike can be terraformed.

Reply:
Ok I tested the diker now.  With a 33% chance the situation is not better then before introducing the dike, as the dike will be removed after every third cycle, and we stilll will remain with beaches which are unreachabele by a new dike.  
So I experimented and reduced it to 10%. now it theoretically survives long enough to be visited more often . However every third cycle the diker builds a new dike (which looks good) but then he may concentrate work on only one of them, or mix them up in any order so the dikes begin to vanish after 20 minutes if it had not been visited yet or 10 minutes after the last visit, resulting again in beaches not transformable anymore.  
So I like to suppose the following changes:
1. call build a dike only after 3 terraforming actions.
2. Let a dike live for at least 30 - 40 minutes if unvisited
3. delete the idle time and final remove from the maybe_remove program
4. reduce the chance to 10 or 15%

that should result in nice line of dikes at the coast with all the terrain around them being slowly transformed. If all triangles have been transformed the player can easily destroy the dike by building a building on it, as it has size none. Due the size none it should be a bit smaller imho as well. If the player does not destroy a dike it will be maintained by delivering logs to it until something is built there to secure the land, this can be explained in the help of the diker, the dike and the dikers house.   
What do you think.

Reply:
> delete the idle time and final remove from the maybe_remove program

This is not possible. When the `maybe_remove` program is started it replaces the main program, and when it runs through it starts over. Then the only statememt ("remove=chance:...") is called again and again with no sleeptime until the removal succeeds. That's why we must phrase this program to effectively replace the main program.

Tweaked the other parameters.

Reply:
ok didn't know that

Reply:
tested the new parameters and it works well. So diking should be fine except an update of the help text to explain the mechanism.  Especially that the player may need to remove the dike if terraforming is finished.  
Thanks for the solution I will link the issue #5459 as being fixed